biz.aQute.resolve/src/biz/aQute/resolve/ProjectResolver.java

```diff
@@ -81,10 +82,14 @@ private String toString(ServiceReference< ? > sr) {
 	private ResolveProcess					resolve		= new ResolveProcess();
 	private Collection<ResolutionCallback>	cbs			= new ArrayList<ResolutionCallback>();
 
-	public ProjectResolver(Project project) {
+	public ProjectResolver(Project project) throws Exception {
 		super(project);
 		getSettings(project);
 		this.project = project;
+
+		Workspace workspace = project.getWorkspace();
+		if (workspace != null)
+			project.addBasicPlugin(new WorkspaceResourcesRepository(workspace));
```

Comment:
That's exactly how the bnd resolve operation works, and I don't disagree
because the projects are a moving target which are not indexed in real-time.

On Aug 31, 2016 10:37 AM, "BJ Hargrave" <notifications@github.com> wrote:

> In biz.aQute.resolve/src/biz/aQute/resolve/ProjectResolver.java
> <https://github.com/bndtools/bnd/pull/1663#discussion_r76999469>:
>
> >  		super(project);
> >  		getSettings(project);
> >  		this.project = project;
> > +
> > +		Workspace workspace = project.getWorkspace();
> > +		if (workspace != null)
> > +			project.addBasicPlugin(new WorkspaceResourcesRepository(workspace));
>
> This seems wrong. Everytime you create a ProjectResolver, you add a new
> WorkspaceResourcesRepository plugin to the workspace.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/bndtools/bnd/pull/1663/files/e24ca8edb8a2cc86c87240547729c03f5819df23#r76999469>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAI9TI0loM4-HdvJtO3b1so13ydxzsnoks5qlZGrgaJpZM4JwIDm>
> .
>